### PR TITLE
New commands

### PR DESCRIPTION
**Pull request by Ijwu**
The framework for the new commands system. I am not hooking it up to chat and packets and what-not because of the modularization efforts being made in another branch. If this PR is accepted I will simply work to make the command module use this code.

Edit: If this PR is accepted I'll take care of any merge conflicts.

### CODE REVIEW

**Comment by QuiCM** [comment]
cc @tylerjwatson @nicatronTg

File: Orion/Commands/CommandManager.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            Commands.Add(comm);
+        }
+
+        public void AddCommand<T1, T2, T3, T4, T5>(string name, Action<T1, T2, T3, T4, T5> commandMethod) where T1 : OrionPlayer
+        {
+            var comm = new Command(name, commandMethod.Method, commandMethod.Target);
+            Commands.Add(comm);
+        }
+
+        public void AddCommand<T1, T2, T3, T4, T5, T6>(string name, Action<T1, T2, T3, T4, T5, T6> commandMethod) where T1 : OrionPlayer
```
Comment (hakusaro) [inline Orion/Commands/CommandManager.cs:72]: Can you clarify the purpose for these overloads @ijwu?

**Comment by Ijwu** [comment]
@nicatronTg The overloads are necessary in order for the command system to know the parameter types of the given method.

For example, if I want to have multiple commands from different overloads of a single method:

``` c#
public void CommandMethod(OrionPlayer ply, string s, int i)
{
    //Stuff
}

public void CommandMethod(OrionPlayer ply, float f)
{
    //Stuff
}

public void RegisterCommands(CommandManager cm)
{
    //Allows for overloads of a method to be commands.
    cm.AddCommand<OrionPlayer, string, int>("commandOne", CommandMethod);
    cm.AddCommand<OrionPlayer, float>("commandOne", CommandMethod);
}
```

Also, without the type arguments then there is no way the compiler can infer the parameter types.

Without the overloads, there is no way to have a command callback with more parameters. 

Six overloads is where I stopped during testing. My initial idea was to have 10 overloads for 9 maximum parameters. 

The first parameter for every command callback must always be an `OrionPlayer` because that is filled in with the player calling the command.

**Comment by Ijwu** [comment]
Closed in favor of moving to OTAPI and using its command system.

**Pull request closed without merging**